### ISSUE TITLE
[docs]: Improve docs regarding AzureIdentity and AzureIdentityBinding

### ISSUE THREAD

**Issue by ThorstenHans** [Azure/kubernetes-keyvault-flexvol#47]
tl;dr: Don't put `AzureIdentity` and `AzureIdentityBinding` in a single yaml file

The `AzureIdentityBinding` depends on `AzureIdentity` obviously. Unfortunately will the drive not do any consistency check here. 

While trying to adopt the KeyVault integration using a PodIdentity, I added both `AzureIdentity` and `AzureIdentityBinding` to a single yaml file. 

The only error I received was that the pod didn't start because the KeyVault volume couldn't be mounted. After some investigation, I recognized that calls to Azure KeyVault were sent without `Authorization` header. 

The KeyVault driver didn't report any real insights... But after a couple of seconds, k8s tried to spin up the Pod again and everything worked as expected. 
I think the behavior is correct, but documentation should point users in the right direction and explicitly mention that `AzureIdentity` and `AzureIdentityBinding` should not belong to a single yaml file.
